File: r-ggplot/plot.R
library(ggplot2)
library(readr)

# Load data
peng <- read_csv("penglings.csv")

# Remove rows with missing values
peng <- na.omit(peng)

# Plot
p <- ggplot(peng, aes(
  x = flipper_length_mm,
  y = body_mass_g,
  color = species,
  size = bill_length_mm
)) +
  geom_point(alpha = 0.7) +
  scale_color_manual(
    values = c(
      Adelie = "orange",
      Chinstrap = "purple",
      Gentoo = "#008080"
    )
  ) +
  scale_x_continuous(
    limits = c(170, 230),
    breaks = seq(170, 230, 10)
  ) +
  scale_y_continuous(
    limits = c(3000, 6000),
    breaks = seq(3000, 6000, 1000)
  ) +
  labs(
    title = "ggplot2",
    x = "Flipper Length (mm)",
    y = "Body Mass (g)",
    size = "Bill Length (mm)",
    color = "species"
  ) +
  theme_minimal() +
	theme(plot.title = element_text(hjust=0.5))


# Save PNG
print(p)
ggsave("ggplot.png", p, width = 8, height = 6, dpi = 300)

File: altair/plot.py
# Altair is concise and decorative, but size scaling is less intuitive than ggplot2

import altair as alt
import pandas as pd

# load data
df = pd.read_csv("penglings.csv")

# some rows have NA values, this code will ignore them
df = df.dropna(subset=[
    "flipper_length_mm",
    "body_mass_g",
    "bill_length_mm",
    "species"
])

chart = alt.Chart(df).mark_circle(opacity=0.7).encode( #opacity 0f 0.7 matches given image
    x=alt.X("flipper_length_mm:Q", title="Flipper Length (mm)",
            scale=alt.Scale(domain=[170,235])),
    y=alt.Y("body_mass_g:Q", title="Body Mass (g)",
            scale=alt.Scale(domain=[2500, 6500])), # this changes the scale in accordance to given image
    color=alt.Color("species:N", title="Species",
                    scale=alt.Scale(
                        range=["orange", "purple", "teal"])), # this gives specific colors to each species
    size=alt.Size(
        "bill_length_mm:Q",
        title="Bill Length (mm)",
        legend=alt.Legend(values=[40,50]) # this specifies the values shown in the legend
    ),
    tooltip=["species", "bill_length_mm", "body_mass_g"]
).properties(
    width=600,
    height=400,
    title="Altair"
)

print("Saving image now...")
chart.save("altair.html") # was not able to figure out how to save to png without downloading extra things

print("done!")
